cylc/flow/network/resolvers.py

```diff
@@ -194,15 +194,18 @@ def node_ids_filter(tokens, state, items) -> bool:
             )
             # match cycle/task/job state
             and (
-                not get_state_from_selectors(item)
+                not (
+                    state is not None
+                    and get_state_from_selectors(item)
+                )
```

Comment:
As written, this logic can be simplified to:
```
(
   state is None
   or get_state_from_selectors(item) is None
)
```
can't it?

(And should it actually be an `and` in the middle?)

Reply:
`and` means the second line doesn't get evaluated right? (which I thought more efficient)
Otherwise yes it could be written that way; `!(A & B)` is equivalent to `(!A or !B)`

Reply:
Nevermind, it works with or too ... 🤦‍♂️

Reply:
fixed

Reply:
Resolved?

Isn't this equivalent to:

```suggestion
                (
                    state is None
                    or get_state_from_selectors(item)
                )
```